File: app/javascript/packs/cards/sonarr/models/episode.js
import format from 'date-fns/format'
import { utcToZonedTime } from 'date-fns-tz'

import Series from './series'

export default class {
  constructor(episode) {
    this.episode = episode
    
    if (this.episode.series)
      this.series = new Series(this.episode.series)
  }

  get title() {
    return this.episode.title
  }

  get poster() {
    return this.series.poster
  }

  get shortAirDate() {
    return this.airDate()
  }

  get downloaded() {
    return this.episode.hasFile
  }

  get number() {
    return `S${this.episode.seasonNumber}E${this.episode.episodeNumber}`
  }

  get seriesTitle() {
    return this.series.title
  }

  airDate(display = 'EEEE K:mmaaaaa') {
    let airDate = utcToZonedTime(new Date(this.episode.airDateUtc))
    return format(airDate, display)
  }

  get url() {
    return `${this.series.url}`
  }

  image(type = 'poster') {
    return this.series.image(type)
  }
}

File: app/services/sonarr/episode.rb
module Sonarr
  module Episode



    def image(type = :poster)
      images = series[:images].select do |image|
        image[:coverType].to_sym == type
      end

      images.first[:url]
    end

    def number
      "S#{episode[:seasonNumber]}E#{episode[:episodeNumber]}"
    end

    def url
      "series/#{series[:titleSlug]}"
    end

  end
end